src/basic/memfd-util.c

```diff
@@ -133,3 +133,34 @@ int memfd_new_and_map(const char *name, size_t sz, void **p) {
 
         return TAKE_FD(fd);
 }
+
+int memfd_new_and_seal(const char *name, const void *data, size_t sz) {
```

Comment:
same as `acquire_data_fd()` with `ACQUIRE_NO_DEV_NULL| ACQUIRE_NO_PIPE`, no?

Reply:
Well sure, but we can reuse this in acquire_data_fd(), and its more descriptive when we just want a memfd and no fallbacks.

Reply:
but then please rework acquire_data_fd() to wrap memfd_new_and_seal()

Reply:
oh, you already do. i am an idiot